src/main/java/ua/ithillet/lesson7/FileNavigator.java

```diff
@@ -0,0 +1,56 @@
+package ua.ithillet.lesson7;
+
+import java.util.stream.Collectors;
+import java.util.*;
+
+
+public class FileNavigator {
+
+    private final Map<String, List<FileBox>> filesLibrary;
+
+    public FileNavigator() {
+        filesLibrary = new HashMap<>();
+    }
+
+    protected Map<String, List<FileBox>> getFilesLibrary() {
+        return filesLibrary;
+    }
+
+    public void add(String pathAsKey, FileBox fileBox) {
+        if (!fileBox.getPath().equals(pathAsKey)) {
+            throw new InconsistencyFileException("Keys used as paths are not equal");
+        }
+        if (!filesLibrary.containsKey(pathAsKey)) {
+            List<FileBox> fileBoxList = new ArrayList<>();
+            filesLibrary.put(pathAsKey, fileBoxList);
+            fileBoxList.add(fileBox);
+        } else {
+            filesLibrary.get(pathAsKey).add(fileBox);
+        }
+    }
+
+    public List<FileBox> find(String pathAsKey) {
+        return filesLibrary.get(pathAsKey);
+    }
+
+    public List<FileBox> filterBySize(Map<String, List<FileBox>> filesLibrary, long maxSize) {
+        return filesLibrary.values().stream()
+                .flatMap(Collection::stream)
+                .filter(fileBox -> fileBox.getFileSize() <= maxSize)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                .toList();
```
З 16-ї java можна так писати. 
https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/util/stream/Stream.html#toList()